core/Services/Calendar/Util/ECalComponent.vala

```diff
@@ -46,11 +46,9 @@ namespace Calendar.Util {
     }
 
     public bool ecalcomponent_is_on_day (ECal.Component component, GLib.DateTime day) {
-#if E_CAL_2_0
-        unowned ICal.Timezone system_timezone = ECal.util_get_system_timezone ();
-#else
-        unowned ICal.Timezone system_timezone = ECal.Util.get_system_timezone ();
-#endif
+        var event_store = Calendar.EventStore.get_default ();
+        unowned ICal.Timezone system_timezone = event_store.system_timezone;
```

Comment:
Do not really need this variable - reference `event_store.system_timezone` directly.